docs/content/basics/ui_testing.md

```diff
@@ -1,4 +1,148 @@
+```
+2. Create test and find our button (with `onView()` method and `withId` assertion by id)
+```kotlin
+    @Test
+    fun pressButtonAndCheckText() {
+        onView(withId(R.id.button))
+    }
+```
+
+!!! Disclaimer
+
+        Important addition - when you are trying to find some `View` that, for example, has same `id` like another View on that screen (that situation usually happens when you are trying to test screens with `RecyclerView`) you can add more `Matchers`
+        Simply place all `Matchers` that apply to single `View` on your screen inside of `allOf()` method
+        It will look like that
+        
+        <br/>@Test
+        <br/>fun someTest() {
+        <br/>&nbsp;&nbsp;&nbsp;onView(allOf(withId(R.id.button), withText("Click me")))
+        <br/>}
+
+3.Then we should perform click on it (with `click()` method from `ViewInteraction`)
+```kotlin
+    @Test
+    fun pressButtonAndCheckText() {
+        onView(withId(R.id.button))
+            .perform(click())
+    }
+```
+4. And finally, let's check that our text is shown (find the `TextView` and assert that it is displayed)
+```kotlin
+    @Test
+    fun pressButtonAndCheckText() {
+        ...
+        onView(withId(R.id.textview))
+            .check(matches(isDisplayed()))
+    }
+```
+Once we run it (run it as usual test, but you should connect real phone or run emulator), we will see standard test result
+![alt text](../images/ui_testing/ui_testing_results.png "First state")
+
+**Final code of our test**
+```kotlin
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
+
+    @get:Rule
+    var activityRule: ActivityScenarioRule<MainActivity>
+            = ActivityScenarioRule(MainActivity::class.java)
+
+    @Test
+    fun pressButtonAndCheckText() {
+        onView(withId(R.id.button))
+            .perform(click())
+
+        onView(withId(R.id.textview))
+            .check(matches(isDisplayed()))
+    }
+}
+```
+
+### What else?
+
+Espresso is quite powerful tool with a lot of abstractions inside it. 
+<br/>But there is a quite famous "cheat-sheet" for Espresso, I hope it will be helpful for everybody.
+
+![alt text](../images/ui_testing/ui_testing_cheat_sheet.jpeg "Cheat sheet")
+
+From that cheat-sheet you can understand which methods you can use for finding the `Views`, which methods for interactions with Views and which methods for checking the data.
+
+### Something like conclusion
+
+To be honest, there is a lot of edge-cases once you are trying to write your own tests. That's one of the reasons why that CookBook was created, so we are suggesting to read next articles to understand how to overcome many of common problems that you will face in Ui testing. <br/>
+Beside that fact, that `Espresso` and `UiAutomator` are the main tools to do Ui testing on Android, you may notice at [Companies experience article](https://android-ui-testing.github.io/Cookbook/adoption/companies_experience/) that almost nobody uses only these tools.<br/>
```

Comment:
I like the point. I would also add something about "new way of testing" due to "Compose emerging".

Additionally, I would reword this though.
Something like "most companies do not use Espresso or UiAutomator *directly*, but through frameworks that wrap them under the hood:
1. Espresso 
   - Kakao
   - Barista
   - Kaspresso
2. UiAutomator
   - Kaspresso (Kautomator)
3. Compose Test Library
   - Kaspresso (since 1.4.0) 
   
Moreover you can write something about "Compose is emerging) and that Kaspresso has also added support for it in 1.4.0